### ISSUE TITLE
shutdown: notify the user of endless umount/swapoff/detach attempts (and perhaps timeout)

### ISSUE THREAD

**Issue by OddBloke** [systemd/systemd#33415]
### Component

other

### Is your feature request related to a problem? Please describe

TL;DR: On a system where an MD device is not removed from `/sys` despite a successful `STOP_ARRAY` ioctl, systemd-shutdown will enter a hard-loop without any evidence this is what's happening.

We recently encountered a kernel bug which meant that the `STOP_ARRAY` ioctl would be accepted for a particular MD device, but would not actually result in the device being removed from `/sys/class/block`. The net result was that `systemd-shutdown` looped infinitely: the apparently-successful ioctl resulted in systemd believing that progress was being made and continuing to attempt detaches (i.e. `changed` was set to `true` in `md_points_list_detach`, so the loop in `main()` wouldn't `break`), and it would then find the MD device in `/sys/class/block` in the next execution of `md_list_get`, etc. (This is almost identical to the behaviour described in #19418 which contains an excellent detailed description of the looping behaviour which I won't reproduce here: the only difference here is which kernel bug is causing the device to remain in `/sys`. Rest assured we aren't running 4.4!)

This issue was made more confusing by the fact that systemd doesn't emit any information about what's going on: the journal is already closed by this point, so you have to enable debug logging to the console to get any information (and then, as it's in a hard-loop, you get _a lot_ of information). It was not obvious that we were stuck in `systemd-shutdown` rather than this being a problem with hardware/firmware/kernel post-systemd until I thought to enable this logging just in case.

### Describe the solution you'd like

I would like `systemd-shutdown` to emit a console message if it is still attempting detach after a certain amount of time/number of loops. This would have allowed us to identify the actual issue we were encountering much sooner. This number could be high, so long as we _eventually_ get some information.

_Ideally_, I'd also like systemd to stop attempting to finalize devices after a certain amount of time and move on with the rest of shutdown. If the `STOP_ARRAY` ioctl _had_ failed, `systemd-shutdown` would just try one more detach, emit a warning and continue anyway, so this would be roughly equivalent. (I certainly appreciate that this might not be desirable: on systems with many nested devices/mounts and/or slow networked storage, this step presumably could legitimately take quite some time.)

### The systemd version you checked that didn't have the feature you are asking for

255
